lib/Extractor/KLEEBuilder.cpp

```diff
@@ -136,7 +140,24 @@ ref<Expr> ExprBuilder::makeSizedArrayRead(unsigned Width, StringRef Name,
       new Array(ArrayNames.makeName(NameStr), 1, 0, 0, Expr::Int32, Width));
   ArrayVars.push_back(Origin);
 
+  std::vector<unsigned> ZeroBits, OneBits;
   UpdateList UL(Arrays.back().get(), 0);
+  if (Origin && Origin->K == Inst::Var && (Origin->KnownZeros != APInt(Width, 0) ||
+                                           Origin->KnownOnes != APInt(Width, 0))) {
+    ref<Expr> KnownZerosExpr = klee::ConstantExpr::create(0, Expr::Bool);
+    ref<Expr> KnownOnesExpr = klee::ConstantExpr::create(0, Expr::Bool);
+    ref<Expr> Var = ReadExpr::create(UL, klee::ConstantExpr::alloc(0, Expr::Int32));
+    // KnownZeros
+    ref<Expr> NotZeros = NotExpr::create(klee::ConstantExpr::alloc(Origin->KnownZeros));
+    ref<Expr> VarOrNotZero = OrExpr::create(Var, NotZeros);
```

Comment:
If the bug isn't here, it is nearby. Build a Debug version of Souper and you will see what I mean. It would have been a good idea to be doing that all along.